src/components/admin/UserInfo.js

```diff
@@ -0,0 +1,107 @@
+import PropTypes from 'prop-types';
+import Button from '@material-ui/core/Button';
+import {Typography, Grid} from '@material-ui/core';
+import IconButton from '@material-ui/core/IconButton';
+import styles from './admin.module.css';
+import ReportIcon from '@material-ui/icons/Report';
+import {
+  Dialog,
+  DialogContent,
+  DialogActions,
+  DialogTitle,
+} from '@material-ui/core';
+import {
+  Table,
+  TableContainer,
+  TableBody,
+  TableHead,
+  TableRow,
+  Paper,
+  TableCell,
+} from '@material-ui/core';
+import EditIcon from '@material-ui/icons/Edit';
+
+// eslint-disable-next-line react/display-name
+const UserInfo = React.forwardRef((props, ref) => {
+  const {user, open, onClose, onEntering} = props;
+
+  return (
+    <Dialog
+      ref={ref}
+      open={open}
+      onClose={onClose}
+      onEntering={onEntering}
+      scroll={'paper'} >
+      <DialogTitle>
+        Membership information
+        <IconButton>
+          <EditIcon />
+        </IconButton>
+      </DialogTitle>
+      <DialogContent>
+        <Typography variant='h6'>
+          {user.name}
+        </Typography>
+        <Grid container spacing={3}>
+          <Grid item>
+            <Paper className={styles.note}>
+              <p>
+                Email: {user.email}
+              </p>
+              <p>
+                Date when joined: {user.joinDate.toDateString()}
+              </p>
+            </Paper>
+          </Grid>
+          <Grid item>
+            <Paper className={styles.note}>
+              <p className={styles.adminNote}>
+                Admin note
+                <IconButton>
+                  <ReportIcon />
+                </IconButton>
+              </p>
+              <p className={styles.adminText}>{user.note}</p>
+            </Paper>
+          </Grid>
+          <Grid item>
+            <Paper>
+              <TableContainer>
+                <Table aria-label='mailing groups' size='small'>
+                  <TableHead>
+                    <TableRow>
+                      <TableCell>Group name</TableCell>
+                      <TableCell>Description</TableCell>
+                    </TableRow>
+                  </TableHead>
+                  <TableBody>
+                    {user.groups.map((group) => (
+                      <TableRow key={group.name}>
+                        <TableCell>{group.name}</TableCell>
+                        <TableCell>{group.description}</TableCell>
+                      </TableRow>
+                    ))}
+                  </TableBody>
+                </Table>
+              </TableContainer>
+            </Paper>
+          </Grid>
+        </Grid>
+      </DialogContent>
+      <DialogActions>
+        <Button onClick={onClose}>Back</Button>
+      </DialogActions>
+    </Dialog>
+  );
+});
+
+UserInfo.propTypes = {
+  user: PropTypes.object.isRequired,
+  open: PropTypes.bool.isRequired,
+  onClose: PropTypes.func.isRequired,
+  onEntering: PropTypes.func.isRequired,
+};
```

Comment:
Could you please explain why do you prefer using PropTypes instead of flow types?
If you prefer PropTypes, that's ok, but we should have a consistent version across the pull requests.
Since PropTypes and flow does the same thing (but proptypes is more verbose), then writing first using proptypes and then creating tickets for refactoring to flow seems like too much work :-)

Reply:
you're right, I have mentioned in a previous pull request that after that PR is closed I'll add flow to the project. It's my fault for not doing that, I completely forgot. I'm adding flow to the project now.

Reply:
Sorry, when I wrote "the same thing" I didn't realised it's not 100% true.
Flow is a static analysis tool which uses a superset of the language, allowing you to add type annotations to all of your code and catch an entire class of bugs at compile time.
PropTypes is a basic type checker which has been patched onto React. It can't check anything other than the types of the props being passed to a given component.